bookstore/clone.py

```diff
@@ -406,3 +406,101 @@ def construct_template_params(self, relpath, fs_clonepath):
     def get_template(self, name):
         """Loads a Jinja template by name."""
         return BOOKSTORE_FILE_LOADER.load(self.settings['jinja2_env'], name)
+
+
+class BookstoreFSCloneAPIHandler(APIHandler):
+    """Handle notebook clone from an accessible file system (local or cloud).
+
+    Provides API handling for ``POST`` and clones a notebook
+    from the specified file system (local or cloud).
+
+    Methods
+    -------
+    initialize(self)
+        Helper to access bookstore settings.
+    post(self)
+        Clone a notebook from the filesystem location specified by the payload.
+    build_content_model(self, content, path)
+        Helper for creating a Jupyter ContentsAPI compatible model.
+
+    See also
+    --------
+    `Jupyter Notebook reference on Custom Handlers <https://jupyter-notebook.readthedocs.io/en/stable/extending/handlers.html#registering-custom-handlers>`_
+    """
+
+    def initialize(self):
+        """Helper to retrieve bookstore setting for the session."""
+        self.bookstore_settings = BookstoreSettings(config=self.config)
+
+    def _get_content(self, path):
+        """Helper for getting content from a specified filepath.
+        
+        Parameters
+        ----------
+        path : str
+           File system path to a notebook
+        """
+        self.log.info(f"Reading content from {path}")
+        if os.path.splitext(path)[1] in [".ipynb", ".jpynb"]:
+            content = self.contents_manager._read_notebook(path)
+        else:
+            content = self.contents_manager._read_file(path, format=None)
+        return content
+
+    @web.authenticated
+    async def post(self):
+        """POST /api/bookstore/fs-clone
+
+        Clone a notebook to the path specified in the payload.
+
+        The payload type for the request should be::
+
+            {
+            "relpath": string,
+            "target_path": string #optional
```

Comment:
Does this successfully capture the idea in a way that would make sense for a python developer?